.github/workflow-scripts/validate-tests.sh

```diff
@@ -31,13 +31,14 @@ then
     testsRemoved=$(cat $TEMP_COMPAR_DIR/diff | grep "<" )
     if [ ! -z "$testsAdded" ]; then
         echo "::error::The following fat(s) need to be added to a category under .github/test-categories: $testsAdded"
+        echo "::warning::Not all fats were run, this build should not be considered complete."
+        exit 0; # Post error to annotation, but do not fail build outright
     fi
 
     if [ ! -z "$testsRemoved" ]; then
         echo "::error::The following fat(s) need to be removed from a category under .github/test-categories: $testsRemoved"
+        exit 1; # Post error to annotation and fail build, otherwise, we will try to run a fat that does not exist.
```

Comment:
I'm not convinced that a missing bucket should prevent all the others from running?

Reply:
The reason I put this validation steps as one of the first things we do is so that the developer can get quick feedback.  If I forget to add/remove a test from a category I want to know as soon as possible so I can correct it and push the change.  If all we do is warn users then they aren't going to realize the mistake until after a 3 hour build and have to rebuild everything again.  

Honestly, I think failing quick is the right way to go.  I'd rather have a build that is building everything I expect or not at all.